.github/workflows/changeset.yml

```diff
@@ -55,8 +71,148 @@ jobs:
           NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
 
       - name: Add Label to CI PR
+        if: ${{ steps.changesets.outputs.hasChangesets == 'true' }}
         run: gh pr edit "$PR_URL" --add-label "ci"
         env:
           PR_URL: ${{ steps.changesets.outputs.pull_request_url }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          
+  snapshot-release:
+    name: Create a snapshot release of a pull request
+    if: ${{ github.repository_owner == 'astrolicious' && github.event.issue.pull_request && (contains(github.event.comment.body, '!preview') || contains(github.event.comment.body, '/preview') || contains(github.event.comment.body, '!snapshot') || contains(github.event.comment.body, '/snapshot')) }}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+      issues: write
+      pull-requests: write
+    steps:
+      - name: "Check if user has write access"
+        uses: "lannonbr/repo-permission-check-action@2.0.2"
+        with:
+          permission: "write"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract the snapshot name from comment body
+        id: getSnapshotName
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const { body } = context.payload.comment;
+            const PREVIEW_RE = /^[!\/](?:preview|snapshot)\s+(\S*)\s*$/gim;
+            const [_, name] = PREVIEW_RE.exec(body) ?? [];
+            if (name) return name;
+
+            const error = 'Invalid command. Expected: "/preview <snapshot-name>"'
+            github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: error,
+            })
+            core.setFailed(error)
+          result-encoding: string
+
+      - name: resolve pr refs
+        id: refs
+        uses: eficode/resolve-pr-refs@main
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ steps.refs.outputs.head_ref }}
+          fetch-depth: 0
+
+      - run: git fetch origin main:main
+
+      - name: Setup Node.js 20.x
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20.x
+
+      - name: Install pnpm
+        run: npm i pnpm@9.5.0 -g
```

Comment:
I think it would be better enable corepack so it uses the exact version from package.json

Reply:
There are a few places that would need to happen, but i am not against this, i just haven't had to much experience setting up corepack within GH actions.  I will note the version in the `package.json` is currently the same.

Reply:
[4093ecb](https://github.com/astrolicious/studiocms/pull/218/commits/4093ecb8988a3f35df3e26af1878dbc5187610e9) done using the pnpm action that references the packageManager option! (i didnt know this action step existed... i like this...)